src/fastify-plugins/utils.js

```diff
@@ -0,0 +1,29 @@
+import { once } from 'node:events'
+
+/**
+ * @param {import('node:http').Server} server
+ * @returns {Promise<string>}
+ */
+export async function getFastifyServerAddress(server) {
+  const address = server.address()
+
+  if (!address) {
+    await once(server, 'listening')
```

Comment:
Again, something from before, which I'm only noticing because I'm reviewing it. I think how it is fine and won't cause issues, but this change makes it a bit more "correct".

Wrapping this  function in a pTimeout leaves the `once(server, 'listening')` hanging. Better to integrate timeout into this function. I only just learnt about `AbortSignal.timeout()` but it's helpful here.

```suggestion
 * @param {import('node:http').Server} server
 * @param {{ timeout?: number }} [options]
 * @returns {Promise<string>}
 */
export async function getFastifyServerAddress(server, { timeout } = {}) {
  const address = server.address()

  if (!address) {
    await once(server, 'listening', { signal: timeout ? AbortSignal.timeout(timeout) : undefined })
```

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/440/commits/3aafbc01327688488971d330e4d98482ac2d93b1